### PR TITLE
GODRIVER-2643 update virtual environment source

### PR DESCRIPTION
**Pull request by prestonvasquez**
GODRIVER-2643

## Summary

replace use of [activate_venv.sh](https://github.com/mongodb-labs/drivers-evergreen-tools/blob/e276a93e6477149362ca85881006a1fae99a9095/.evergreen/csfle/activate_venv.sh) with [activate-kmstlsvenv.sh](https://github.com/mongodb-labs/drivers-evergreen-tools/blob/e276a93e6477149362ca85881006a1fae99a9095/.evergreen/csfle/activate-kmstlsvenv.sh).

## Background & Motivation

The activate-kmstlsvenv.sh is a hygienic alternatives to activate_venv.sh that does not leak variables into the environment and does not depend on environment variables for its behavior.

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -605,8 +621,11 @@ functions:
   run-valid-ocsp-server:
     - command: shell.exec
       params:
+        shell: "bash"
         script: |
           cd ${DRIVERS_TOOLS}/.evergreen/ocsp
+          . ./activate-ocspvenv.sh
+
           ${PYTHON3_BINARY} -m venv ./venv
```
Comment (benjirewis) [inline .evergreen/config.yml:629]: What virtual environment were these OCSP tasks using before the inclusion of this `activate-ocspvenv` script?

**Review by benjirewis — COMMENTED** [review]
Looking good; nice to get rid of a lot of the python binary logic.

File: .evergreen/config.yml (hunk 1 above)
Comment (prestonvasquez) [reply .evergreen/config.yml:629]: I think it uses the virtual environment from `./venv/${VENV_BIN_DIR|bin}/python`.

File: .evergreen/config.yml [hunk 2]
```diff
@@ -170,6 +170,22 @@ functions:
           # initialize submodules
           git submodule init
           git submodule update
+    - command: shell.exec
+      params:
+        working_dir: src/go.mongodb.org/mongo-driver
+        shell: "bash"
+        script: |
+          . ${DRIVERS_TOOLS}/.evergreen/utils.sh
```
Comment (kevinAlbs) [inline .evergreen/config.yml:178]:
```suggestion
          . ${DRIVERS_TOOLS}/.evergreen/venv-utils.sh
```

`venv-utils.sh` is the hygienic alternative.

File: .evergreen/config.yml [hunk 3]
```diff
@@ -605,8 +621,11 @@ functions:
   run-valid-ocsp-server:
     - command: shell.exec
       params:
+        shell: "bash"
         script: |
           cd ${DRIVERS_TOOLS}/.evergreen/ocsp
+          . ./activate-ocspvenv.sh
+
           ${PYTHON3_BINARY} -m venv ./venv
           ./venv/${VENV_BIN_DIR|bin}/pip3 install -r mock-ocsp-responder-requirements.txt
```
Comment (kevinAlbs) [inline .evergreen/config.yml:630]:
```suggestion
```

`activate-ocspvenv.sh` creates the virtual environment and installs dependencies.

File: .evergreen/config.yml [hunk 4]
```diff
@@ -624,8 +643,11 @@ functions:
   run-revoked-ocsp-server:
     - command: shell.exec
       params:
+        shell: "bash"
         script: |
           cd ${DRIVERS_TOOLS}/.evergreen/ocsp
+          . ./activate-ocspvenv.sh
+
           ${PYTHON3_BINARY} -m venv ./venv
           ./venv/${VENV_BIN_DIR|bin}/pip3 install -r mock-ocsp-responder-requirements.txt
```
Comment (kevinAlbs) [inline .evergreen/config.yml:652]:
```suggestion
```

File: .evergreen/config.yml [hunk 5]
```diff
@@ -645,8 +667,11 @@ functions:
   run-valid-delegate-ocsp-server:
     - command: shell.exec
       params:
+        shell: "bash"
         script: |
           cd ${DRIVERS_TOOLS}/.evergreen/ocsp
+          . ./activate-ocspvenv.sh
+
           ${PYTHON3_BINARY} -m venv ./venv
           ./venv/${VENV_BIN_DIR|bin}/pip3 install -r mock-ocsp-responder-requirements.txt
```
Comment (kevinAlbs) [inline .evergreen/config.yml:676]:
```suggestion
```

File: .evergreen/config.yml [hunk 6]
```diff
@@ -664,8 +689,11 @@ functions:
   run-revoked-delegate-ocsp-server:
     - command: shell.exec
       params:
+        shell: "bash"
         script: |
           cd ${DRIVERS_TOOLS}/.evergreen/ocsp
+          . ./activate-ocspvenv.sh
+
           ${PYTHON3_BINARY} -m venv ./venv
           ./venv/${VENV_BIN_DIR|bin}/pip3 install -r mock-ocsp-responder-requirements.txt
```
Comment (kevinAlbs) [inline .evergreen/config.yml:698]:
```suggestion
```

**Review by kevinAlbs — APPROVED** [review]

**Review by benjirewis — APPROVED** [review]
Kevin's suggestions handle my concerns about the `activate-ocspvenv` script. LGTM.

**Pull request merged**